tests/pysharedutils/dicts_tests.py

```diff
@@ -281,3 +281,100 @@ def test_get_dict_properties_without_strict_and_invalid_property(self):
             'comment.user.username'
         )
         assert_equal(output, {'comment.user.username': None})
+
+
+class TestMapDictKeys:
+
+    def test_map_dict_keys(self):
+        obj = {
+            'first_name': 'Joe',
+            'middle_name': 'P',
+            'last_name': 'Smith'
+        }
+        map_obj = {
+            'first_name': 'given_name',
+            'last_name': 'surname',
+            'mis_spelled_key': 'mis_spelled_value'
+        }
+        output = pysharedutils.map_dict_keys(obj, map_obj)
+        expected_output = {
+            'given_name': 'Joe',
+            'middle_name': 'P',
+            'surname': 'Smith'
+        }
+        assert_equal(output, expected_output)
+
+    def test_map_dict_keys_with_one_level(self):
+        obj = {
+            'first_name': 'Joe',
+            'user': {
+                'username': 'joe@example.com'
+            }
+        }
+        map_obj = {
+            'first_name': 'given_name',
+            'user.username': 'email',
```

Comment:
Maybe you are confused by the choice of example I chose for the test? mapping `username` to `email`. 
The map dict is basically of the format `{'path_to_the_key': 'new_key_name'}`.